rfd/0046-database-access-config.md

```diff
@@ -64,14 +66,269 @@ $ clusterawsadm bootstrap iam print-policy --document AWSIAMManagedPolicyControl
 
 Teleport has a `teleport configure` command that generates a sample config file.
 
-Database agent bootstrap commands will reside under `teleport db configure`
-family of subcommands. This aligns with our strategy of placing commands related
-to a particular service under its own subcommand namespace.
+Database agent bootstrap commands will reside under teleport db configure family
+of subcommands. This aligns with our strategy of placing commands related to a
+particular service under its subcommand namespace. It will provide subcommands
+for generating configuration files and bootstrapping database agents’ necessary
+configurations, for example, setting up IAM permissions for agents to proxy
+AWS-hosted databases.
+
+### Create config subcommand
+
+Similar to `teleport configure` but focused on configurations for Database
+agents. This command will generate a configuration file. It can also provide
+some examples of configuration, for instance, self-hosted databases setup.
+
+There will be flags to enable features like Aurora/RDS auto-discovery and
+dynamic registration. Each sample will show the required options with some
+explanation on how to fill them and documentation links.
+
+It will also provide flags to configure a specific database (similar to the
+flags present on the `teleport db start` command). When executed with these
+flags, the samples will not be included, only the static configuration of a
+single database.
+
+```bash
+teleport db configure create [--auth-server] [--token=] [--version=] [--enable-rds-auto-discovery] [--enable-dynamic-registration] [--output=] [--ca-pin] [--name] [--protocol] [--uri]
+```
+
+| Flag                            | Description |
+| ------------------------------- | ----------- |
+| `--auth-server`                 | Address of the auth server. |
+| `--token`                       | Invitation token to register with an auth server |
+| `--version`                     | Teleport configuration version. |
+| `--enable-rds-auto-discovery`   | Enables Aurora/RDS auto-discovery. |
```

Comment:
How will the user specify auto-discovery parameters such as types/regions/tags? For tags we could probably default to wildcard, but I think we need some way to provide types ("rds", "redshift", etc.) and regions.

Reply:
I'm thinking of having different flags to enable auto-discovery, and some additional flags to customize them. In this way each "auto-discovery" can have its own options. For example:

```bash
# Enables Redshift and RDS auto-discovery on us-west-1 and us-west-2
$ teleport db configure create \
    --enable-redshift-auto-discovery \
    --redshift-auto-discovery-regions us-west-1,us-west-2 \
    --enable-rds-auto-discovery \
    --rds-auto-discovery-regions us-west-1,us-west-2

# Set tags for RDS auto discovery.
$ teleport db configure create \
    --enable-rds-auto-discovery \
    --rds-auto-discovery-tag owner=me \
    --rds-auto-discovery-tag department=IT \
```

What do you think?

Reply:
Thinking more about this, we probably don't want to overload the CLI interface of the command with too many flags. How about we get rid of the `--enable-auto-discovery` boolean switch and only keep the "regions" flag:

```
--discover-rds=us-east-1 --discover-rds=us-central-1 --discover-redshift=us-west-2
```

This will generate a config with enabled auto-discovery with wildcard labels. If people want to tweak them, they can do it in the config.

Reply:
That's good a way to go, do you think we still need the tags flags?